### PR TITLE
[runtime/sgen] Implement reflection-only private api to get referring objects

### PR DESCRIPTION
**Pull request by alexanderkyte**
In order to support the work of a GSOC student, I'm exposing a function that can be used to get the objects that we know contain references to a given object. It's designed to be reflected out of the System.GC-contiaining assembly. This design was inspired by Microsoft:

https://github.com/Microsoft/referencesource/blob/master/mscorlib/system/threading/Tasks/TaskScheduler.cs#L515-L529

### CODE REVIEW

File: mcs/class/corlib/System/GC.cs [hunk 1]
```diff
@@ -156,4 +158,18 @@ public static class GC
 
 		internal static readonly object EPHEMERON_TOMBSTONE = get_ephemeron_tombstone ();
 	}
+
+	[Serializable]
+	[StructLayout(LayoutKind.Sequential)]
+	internal class ReferringObject
```
Comment (marek-safar) [inline mcs/class/corlib/System/GC.cs]: Move both new types to individual cs files

**Comment by saper** [comment]
build

**Comment by lewurm** [comment]
is there still interest to merge this or should it be closed?

**Comment by mattzink** [comment]
This would be extremely useful for debugging managed leaks, so I would vote this gets merged

**Comment by lewurm** [comment]
@alexanderkyte can you have a look and rebase your changes?

**Pull request closed without merging**